packages/@n8n/task-runner-python/src/logs.py

```diff
@@ -70,3 +76,23 @@ def setup_logging():
     logger.addHandler(stream_handler)
 
     logging.getLogger("websockets.client").setLevel(logging.DEBUG)
+
+
+def format_print_args(*args) -> list[str]:
+    """Format `print` arguments for browser console output, handling circular references."""
+
+    formatted_args = []
+    for arg in args:
+        if isinstance(arg, dict) and CIRCULAR_REFERENCE_PLACEHOLDER_KEY in arg:
+            formatted_args.append(f"[Circular {arg['__type__']}]")
```

Comment:
Why not do this formatting in the task executor already?

Reply:
I think I thought logs formatting belonged to `logs.py` rather than `task_executor.py`. Moved anyway.